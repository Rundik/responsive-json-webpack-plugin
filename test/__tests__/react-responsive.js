const React = require('react');
const ResponsiveImage = require('../../src/react.tsx');
const shallow = require('enzyme').shallow;

test('snapshot', () => {
    const wrapper = shallow(
        <ResponsiveImage
            image={{
                src: 'test.jpg'
            }}
            imagePath="root/"
        />
    );
    expect(wrapper).toMatchSnapshot();

    const wrapper1 = shallow(
        <ResponsiveImage
            image={{
                src: 'test.jpg'
            }}
            imagePath="root/"
            onClick={() => undefined}
        />
    );
    expect(wrapper1).toMatchSnapshot();

    const wrapper2 = shallow(
        <ResponsiveImage
            image={{
                src: 'sample-8.png',
                alt: 'Sample ALT',
                sizes:
                    '(max-width: 56.25em) 20vw, (max-width: 37.5em) 30vw, 300px',
                srcset: [
                    {
                        src: 'sample-8x8',
                        size: 8
                    }
                ]
            }}
        />
    );
    expect(wrapper2).toMatchSnapshot();

    const wrapper3 = shallow(
        <ResponsiveImage
            image={{
                src: 'sample-8.png',
                sizes:
                    '(max-width: 56.25em) 20vw, (max-width: 37.5em) 30vw, 300px',
                srcset: [
                    {
                        src: 'sample-8x8',
                        size: 8
                    }
                ],
                sources: [
                    {
                        media: '(max-width: 37.5em)',
                        sizes: '20vw',
                        srcset: [
                            {
                                src: 'potato.png',
                                size: 16
                            },
                            {
                                src: 'sample-10-x16',
                                size: 16
                            }
                        ]
                    }
                ]
            }}
        />
    );
    expect(wrapper3).toMatchSnapshot();
});
