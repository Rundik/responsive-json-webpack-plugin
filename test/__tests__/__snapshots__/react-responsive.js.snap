// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<img
  alt="test.jpg"
  src="root/test.jpg"
/>
`;

exports[`snapshot 2`] = `
<img
  alt="test.jpg"
  onClick={[Function]}
  src="root/test.jpg"
/>
`;

exports[`snapshot 3`] = `
<img
  alt="Sample ALT"
  sizes="(max-width: 56.25em) 20vw, (max-width: 37.5em) 30vw, 300px"
  src="/sample-8.png"
  srcSet="/sample-8x8 8w"
/>
`;

exports[`snapshot 4`] = `
<picture>
  <source
    key="0"
    media="(max-width: 37.5em)"
    sizes="20vw"
    srcSet="/potato.png 16w, /sample-10-x16 16w"
  />
  <img
    alt="sample-8.png"
    sizes="(max-width: 56.25em) 20vw, (max-width: 37.5em) 30vw, 300px"
    src="/sample-8.png"
    srcSet="/sample-8x8 8w"
  />
</picture>
`;
